### PR TITLE
Add more comprehensive documentation on supported date formats

### PR DESCRIPTION
**Pull request by andygrove**
Documentation improvement

### CODE REVIEW

**Review by sameerz — DISMISSED** [review]
Approving for the grammar and formatting, I have not verified the supported formats.

**Comment by razajafri** [comment]
This looks good but would be great if we can somehow leverage the TypeChecks to automatically document this in the compatibility.md

File: docs/compatibility.md [hunk 1]
```diff
@@ -235,27 +235,51 @@ from one run to another if the ordering is ambiguous on a window function too.
 ## Parsing strings as dates or timestamps
 
 When converting strings to dates or timestamps using functions like `to_date` and `unix_timestamp`,
-only a subset of possible formats are supported on GPU with full compatibility with Spark. The
-supported formats are:
+the specified format string will fall into one of three categories:
+
+- Supported on GPU, with compatibility with Spark
+- Supported on GPU, but may produce different results to Spark
```
Comment (jlowe) [inline docs/compatibility.md:241]:
```suggestion
- Supported on GPU and 100% compatible Spark
- Supported on GPU but may produce different results to Spark
```

**Comment by razajafri** [comment]
Can we create a follow-on issue to handle that for future triage?

**Review by razajafri — DISMISSED** [review]

File: docs/compatibility.md [hunk 2]
```diff
@@ -235,27 +235,51 @@ from one run to another if the ordering is ambiguous on a window function too.
...
+- Unsupported on GPU
+
+The supported formats are:
```
Comment (jlowe) [inline docs/compatibility.md:244]: Which version of "supported" are these?  The above bullets list two forms of supported, and it's ambiguous whether these are just the 100% supported or both forms of supported.

File: docs/compatibility.md [hunk 3]
```diff
@@ -235,27 +235,51 @@ from one run to another if the ordering is ambiguous on a window function too.
...
+- `dd/MM`
 
-Other formats may result in incorrect results and will not run on the GPU by default. Some
-specific issues with other formats are:
+Examples of supported formats that may produce different results are:
 
-- Spark supports partial microseconds but the plugin does not
-- The plugin will produce incorrect results for input data that is not in the correct format in
-some cases
+- Use of two digit years `yy` may use different century on GPU compared to CPU
```
Comment (jlowe) [inline docs/compatibility.md:259]:
```suggestion
- Use of a two digit year, i.e.: `yy`, may use a different century on GPU compared to CPU
```

**Comment by jlowe** [comment]
> This looks good but would be great if we can somehow leverage the TypeChecks to automatically document this in the compatibility.md

TypeChecks check types, not values, so IMO that's not an appropriate mechanism for documenting it.  It may be worth adding a `withPsNote` if one doesn't already exist to point to this document for any operation parsing formats.

**Review by jlowe — APPROVED** [review]

**Comment by andygrove** [comment]
build

**Pull request merged**